### PR TITLE
samples: nrf9160: modem_shell: fix LwM2M initialization

### PR DESCRIPTION
**Pull request by tokangas**
There was a race condition with nRF modem library and LwM2M initialization, sometimes causing the reading of IMEI to fail. `cloud_lwm2m_init()` is now called using `NRF_MODEM_LIB_ON_INIT()` to make sure modem library has been initialized before trying to read the IMEI.

### CODE REVIEW

**Review by jhirsi — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c [hunk 1]
```diff
@@ -163,14 +175,12 @@ static int cloud_lwm2m_init(const struct device *unused)
 	ret = modem_info_init();
 	if (ret < 0) {
 		printk("LwM2M: Unable to init modem_info (%d)", ret);
-		return ret;
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c:166]: How does the code work if modem_info_init() fails and we request IMEI then?

File: samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c [hunk 2]
```diff
@@ -163,14 +175,12 @@ static int cloud_lwm2m_init(const struct device *unused)
...
 	}
 
 	/* Query IMEI. */
 	ret = modem_info_string_get(MODEM_INFO_IMEI, imei_buf, sizeof(imei_buf));
 	if (ret < 0) {
 		printk("LwM2M: Unable to get IMEI");
-		return ret;
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c:173]: How does the code work if we don't get IMEI and we use it in endpoint_name?

**Review by trantanen — COMMENTED** [review]
Asking because I don't know how lwm2m should be constructed.

File: samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c (hunk 2 above)
Comment (tokangas) [reply samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c:173]: And same here.

**Review by trantanen — APPROVED** [review]

**Comment by tokangas** [comment]
Need to wait for upmerge to be finished because of a merge conflict.

**Pull request merged**